include/CFL/CFLAlias.h

```diff
@@ -57,39 +59,37 @@ class CFLAlias : public PointerAnalysis
     /// Start Analysis here (main part of pointer analysis).
     virtual void analyze()
     {
-        PointerAnalysis::initialize();
-        GrammarBuilder * gReader = new GrammarBuilder(Options::GrammarFilename);
-        CFGNormalizer *normalizer = new CFGNormalizer();
-
-        graph = new CFLGraph();
+        GrammarBuilder gReader = GrammarBuilder(Options::GrammarFilename);
+        CFGNormalizer normalizer = CFGNormalizer();
+        CFLGraphBuilder cflGraphBuilder = CFLGraphBuilder();
+        CFLChecker cflChecker = CFLChecker();
+        /// Assume Read From Const Graph, associate label symbol is hard coded
         if (Options::GraphIsFromDot == false)
         {
-            // Maybe could put in SVFIR Class
-            // In memory Graph does not have string type label
-            Map<std::string, SVF::CFLGraph::Symbol> ConstMap =  {{"Addr",0}, {"Copy", 1},{"Store", 2},{"Load", 3},{"Gep", 4},{"Vgep", 5},{"Addrbar",6}, {"Copybar", 7},{"Storebar", 8},{"Loadbar", 9},{"Gepbar", 10},{"Vgepbar", 11}};
+            PointerAnalysis::initialize();
+            Map<std::string, SVF::CFLGraph::Symbol> ConstMap =  {{"Addr",0}, {"Copy", 1},{"Store", 2},{"Load", 3},{"Gep_i", 4},{"Vgep", 5},{"Addrbar",6}, {"Copybar", 7},{"Storebar", 8},{"Loadbar", 9},{"Gepbar_i", 10},{"Vgepbar", 11}};
+            GrammarBase *generalGrammar = gReader.build(ConstMap);
             ConstraintGraph *consCG = new ConstraintGraph(svfir);
-            svfir->dump("SVFIR");
-            // Can be put in build, copy from general graph
-            graph->label2SymMap = ConstMap;
-            graph->buildBigraph(consCG);
-            graph->dump("PAG");
+            graph = cflGraphBuilder.buildBigraph(consCG);
+            cflChecker.check(generalGrammar, graph);
```

Comment:
The CFL checker should be able to check any grammar with a graph including the grammar after normalization.

Reply:
GramGraphChecker is currently receiving the GrammarBase should be also able received other child class like cflgrammar.

Reply:
Any way to check whether the normalized grammar is correct and consistent with the graph?